extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -189,12 +190,15 @@ export class AmpAnalytics extends AMP.BaseElement {
             trigger['selector'] = this.expandTemplate_(trigger['selector'],
                 trigger, /* arg*/ undefined, /* arg */ undefined,
                 /* arg*/ false);
-            this.instrumentation_.addListener(
-                trigger, this.handleEvent_.bind(this, trigger), this.element);
-
+            this.instrumentationPromise_.then(instrumentation => {
```

Comment:
would it be simpler to set the value of instrumentation service at the end of layoutcallback. So something like this:

```
     return instrumentationServiceForDoc(this.getAmpDoc()))
         .then(instrumentation => {this.instrumentation_ = instrumentation;})
         .then(() => this.consentPromise_)
         .then(this.fetchRemoteConfig_.bind(this))
         .then(this.onFetchRemoteConfigSuccess_.bind(this));
```